Draft release notes up to zed/2864 (#2872)
brimdata/zed: CHANGELOG.md

* Fix an issue where `cut(.)` could cause a `slice bounds out of range` panic (#2107)
* Add `is()`, `fields()`, and `exists()` functions (#2131)
* Add auto-detection of ZSON format (#2123)

> * Fix an issue where [`cut`](https://github.com/brimdata/zed/tree/main/docs/language/operators#cut) to the root would exit if the referenced column was missing from a record (#2121)

Reviewer: We primarily use the term "field" elsewhere.
```suggestion
* Fix an issue where [`cut`](https://github.com/brimdata/zed/tree/main/docs/language/operators#cut) to the root would exit if the referenced field was missing from a record (#2121)
```

* Minimize the amount of surrounding context shown when reporting parse errors (#2864)
* Field assignments in [`join`](https://github.com/brimdata/zed/blob/main/docs/language/operators/README.md#join) now behave like [`cut`](https://github.com/brimdata/zed/blob/main/docs/language/operators/README.md#cut) instead of [`pick`](https://github.com/brimdata/zed/blob/main/docs/language/operators/README.md#pick) (#2868)
* Add more background/context to Zed top-level language [README](https://github.com/brimdata/zed/blob/main/docs/language/README.md) (#2866 #2878, #2901)
* Unify `from`, `split`, and `switch` syntax to the forms shown [here](https://github.com/brimdata/zed/tree/main/docs/language#directed-acyclic-flow-graphs) (#2871, #2896)
* Shapers can now cast values of the `null` type to any type (e.g., arrays or records) (#2882)
* Fix an issue where [`join`](https://github.com/brimdata/zed/blob/main/docs/language/operators/README.md#join) was failing to match on values of comparable types (e.g., `string` and `bstring`) (#2880, #2884)
* Shapers can now cast a value to a `union` type (#2881)

> * Introduce alternate `switch()` syntax (#2888, #3004)

Reviewer: We refer to this as `switch` (without parentheses) above.
```suggestion
* Introduce alternate `switch`)` syntax (#2888, #3004)
```

* When [`fuse`](https://github.com/brimdata/zed/tree/main/docs/language/operators#fuse) encounters a field with the same name but different types, it now creates one field of `union` type rather than separate, uniquely-named fields (#2885, #2886)
* Fix an issue where [`fuse`](https://github.com/brimdata/zed/tree/main/docs/language/operators#fuse) would consume too much memory when fusing many types (#2897, #2899)

> * Emphasize in the [`sort`](https://github.com/brimdata/zed/tree/main/docs/language/operators#sort) documentation that its output can be non-deterministic in the absence of an explicit field-list (#2902)

Reviewer: ```suggestion
* Emphasize in the [`sort`](https://github.com/brimdata/zed/tree/main/docs/language/operators#sort) documentation that its output can be non-deterministic in the absence of an explicit field list (#2902)
```

* Remove the space separator before decorator in ZSON `-z` output (#2911)
* Fix an issue where handling of record alises caused a failure to shape Zeek NDJSON data (#2904)
* Fix an issue where posting garbage input data to a pool caused an HTTP 500 response (#2924)
* Fix an issue where reading a ZNG file and outputting as CSV caused a deadlock (#2929)
* In a `from` clause, `range` is now used instead of `over` to specify a range scan over a data source (#2943)
* Fix a Zed language issue with parsing parenthesized search terms (#2951)
* Column headers in `-f table` outputs now reflect the case of the field name rather than always being uppercase (#2964)
* Reserved words in the Zed language can now be used in more places (e.g., field name references) without risk of collisions that would require escaping (#2968)
* Zed CLI tools now send human-readable ZSON by default if output is to a terminal, otherwise binary ZNG (#2979, #2985)
* Temporary directories for spill-to-disk operations now are prefixed with `zed-spill-` rather than `zq-spill-` (#2980)

> * The [`put`](https://github.com/brimdata/zed/tree/main/docs/language/operators#put) operator keyword is now optional (e.g., can write`x:=1` instead of `put x:=1`) (#2967, #2986, #3043)

Reviewer: ```suggestion
* The [`put`](https://github.com/brimdata/zed/tree/main/docs/language/operators#put) operator keyword is now optional (e.g., can write `x:=1` instead of `put x:=1`) (#2967, #2986, #3043)
```

* Fix an issue where a [`put`](https://github.com/brimdata/zed/tree/main/docs/language/operators#put) on a nested record with an alias triggered a panic (#2990)
* Fix an issue where temporary spill-to-disk directories were not being deleted upon exit (#3009, #3010)
* Fix a ZSON issue with `union` types with alias decorators (#3015, #3016)
* The ZSON format has been changed such that integer type IDs are no longer output (#3017)
* Update the reference Zed shaper for Zeek ([shaper](https://github.com/brimdata/zed/blob/main/zeek/shaper.zed), [docs](https://github.com/brimdata/zed/blob/main/zeek/Shaping-Zeek-NDJSON.md)) to reflect changes in Zeek release v4.1.0 (#3021)

> * Fix an issue where backslash escapes in Zed regexps were not being accepted (#3040)

Reviewer: ```suggestion
* Fix an issue where backslash escapes in Zed regular expressions were not accepted (#3040)
```
